### PR TITLE
Enable `no_useless_concat_operator`

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Depends on #6674 

```console
$ vendor/bin/php-cs-fixer describe no_useless_concat_operator
Description of no_useless_concat_operator rule.
There should not be useless concat operations.

Fixer is configurable using following option:
* juggle_simple_strings (bool): allow for simple string quote juggling if it results in more concat-operations merges; defaults to false

Fixing examples:
 * Example #1. Fixing with the default configuration.
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -1,2 +1,2 @@
    <?php
   -$a = 'a'.'b';
   +$a = 'ab';

   ----------- end diff -----------

 * Example #2. Fixing with configuration: ['juggle_simple_strings' => true].
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ -1,2 +1,2 @@
    <?php
   -$a = 'a'."b";
   +$a = "ab";

   ----------- end diff -----------
```

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by kenjis — APPROVED** [review]
Please rebase.

**Review by kenjis — APPROVED** [review]

File: tests/system/Images/GDHandlerTest.php [hunk 1]
```diff
@@ -413,7 +413,7 @@ public function testImageConvert()
     public function testImageReorientLandscape()
     {
         for ($i = 0; $i <= 8; $i++) {
-            $source = $this->origin . 'EXIFsamples/landscape_' . '' . $i . '.jpg';
```
Comment (MGatner) [inline tests/system/Images/GDHandlerTest.php:416]: 🤔

File: tests/system/HTTP/MessageTest.php [hunk 2]
```diff
@@ -156,7 +156,7 @@ public function testAppendBody()
 
         $this->message->appendBody('\n');
 
-        $this->assertSame('moo' . '\n', $this->message->getBody());
```
Comment (MGatner) [inline tests/system/HTTP/MessageTest.php:159]: This one almost got me 😆

**Review by MGatner — APPROVED** [review]
Actually got some changes out of this one! Nice rule.

**Pull request merged**